doc/explanation/api/reactivity.md

```diff
@@ -0,0 +1,177 @@
+# Reactivity in Panel
+
+Panel's approach to expressing interactivity and managing state differs from many other frameworks and offers multiple APIs with different affordances. In this overview section we will try to convey the thinking that led to the different APIs in Panel, providing you with context on when to use which and why. In the process we will try to unpack a number of concepts that are central to most UI frameworks and how these concepts translate into Panel's unique approach.
```

Comment:
I think I learn some things from this guide. But I don't think I have really learned anything about when to which and why. Make I can deduce it from what is written. But its not clear to me. Can it be summarized in a decission tree or table?

Reply:
Think I'll write the second guide on functions vs classes and then figure out how to give that guidance.